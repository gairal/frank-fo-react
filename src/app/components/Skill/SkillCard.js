import React from 'react';
import PropTypes from 'prop-types';
import {
  Grid,
  Card,
  CardContent,
  ListItem,
  ListItemText,
  ListItemIcon,
  Typography,
} from '@material-ui/core';
import SkillLevel from './SkillLevel';

const SkillCard = ({ category }) => {
  const skillItem = skill => {
    let skillTitle = skill.name;
    if (skill.description) {
      skillTitle += ` | ${skill.description}`;
    }

    const progress = skill.level || 0;

    return (
      <Grid item xs={12} md={6} lg={4} key={skill.name}>
        <ListItem key={skill.name}>
          <ListItemIcon>
            <SkillLevel level={progress} />
          </ListItemIcon>
          <ListItemText
            primary={skillTitle}
            style={{
              flex: '1 1 0',
            }}
          />
        </ListItem>
      </Grid>
    );
  };

  return (
    <Card>
      <CardContent>
        <Typography variant="headline" component="h2">
          {category.name}
        </Typography>
        <Grid container>
          {category.skills ? category.skills.map(skillItem) : null}
        </Grid>
      </CardContent>
    </Card>
  );
};

SkillCard.propTypes = {
  category: PropTypes.shape().isRequired,
};

export default SkillCard;
