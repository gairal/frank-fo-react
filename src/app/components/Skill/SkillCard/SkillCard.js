import React from 'react';
import PropTypes from 'prop-types';
import Card, { CardContent } from 'material-ui/Card';
import List, { ListItem, ListItemText, ListItemIcon } from 'material-ui/List';
import Typography from 'material-ui/Typography';
import SkillLevel from '../SkillLevel';

const SkillCard = props => (
  <Card>
    <CardContent>
      <Typography type="headline" component="h2">
        {props.category.name}
      </Typography>
      <List>
        {props.category.skills.map((skill) => {
          let skillTitle = skill.name;
          if (skill.description) {
            skillTitle += ` | ${skill.description}`;
          }

          const progress = skill.level || 0;

          return (
            <ListItem key={skill.name}>
              <ListItemIcon>
                <SkillLevel level={progress} />
              </ListItemIcon>
              <ListItemText primary={skillTitle} />
            </ListItem>
          );
        })}
      </List>
    </CardContent>
  </Card>
);

SkillCard.propTypes = {
  category: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default SkillCard;
