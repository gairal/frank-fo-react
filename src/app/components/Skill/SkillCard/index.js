import skillCard from './SkillCard';

export default skillCard;
