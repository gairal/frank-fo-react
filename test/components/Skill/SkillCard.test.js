import React from 'react';
import renderer from 'react-test-renderer';
import SkillCard from '@/components/Skill/SkillCard';

describe('components > Skill > SkillCard', () => {
  it('Renders correctly default', () => {
    const component = renderer.create(<SkillCard category={{ name: 'test' }} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders correctly with categories', () => {
    const component = renderer.create(<SkillCard
      category={{
        name: 'test',
        skills: [
          {
            name: 'test1',
            description: 'test',
            level: 77,
          },
          {
            name: 'test2',
          },
        ],
      }}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
