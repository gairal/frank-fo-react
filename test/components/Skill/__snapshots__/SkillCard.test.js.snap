// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Skill > SkillCard Renders correctly default 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
>
  <div
    className="MuiCardContent-root-29"
  >
    <h2
      className="MuiTypography-root-30 MuiTypography-headline-35"
    >
      test
    </h2>
    <div
      className="MuiGrid-container-55"
    />
  </div>
</div>
`;

exports[`components > Skill > SkillCard Renders correctly with categories 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
>
  <div
    className="MuiCardContent-root-29"
  >
    <h2
      className="MuiTypography-root-30 MuiTypography-headline-35"
    >
      test
    </h2>
    <div
      className="MuiGrid-container-55"
    >
      <div
        className="MuiGrid-item-56 MuiGrid-grid-xs-12-95 MuiGrid-grid-md-6-117 MuiGrid-grid-lg-4-129"
      >
        <li
          className="MuiListItem-root-152 MuiListItem-default-155 MuiListItem-gutters-159"
          disabled={false}
        >
          <div
            aria-valuenow={0}
            className="MuiCircularProgress-root-163 MuiCircularProgress-colorPrimary-166 MuiCircularProgress-static-164"
            role="progressbar"
            style={
              Object {
                "height": 40,
                "transform": "rotate(-90deg)",
                "width": 40,
              }
            }
          >
            <svg
              className="MuiCircularProgress-svg-168"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle-169 MuiCircularProgress-circleStatic-170"
                cx={44}
                cy={44}
                fill="none"
                r={19.5}
                strokeWidth={5}
                style={
                  Object {
                    "strokeDasharray": "122.522",
                    "strokeDashoffset": "122.522px",
                  }
                }
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root-172"
            style={
              Object {
                "flex": "1 1 0",
              }
            }
          >
            <span
              className="MuiTypography-root-30 MuiTypography-subheading-37 MuiListItemText-primary-175"
            >
              test1 | test
            </span>
          </div>
        </li>
      </div>
      <div
        className="MuiGrid-item-56 MuiGrid-grid-xs-12-95 MuiGrid-grid-md-6-117 MuiGrid-grid-lg-4-129"
      >
        <li
          className="MuiListItem-root-152 MuiListItem-default-155 MuiListItem-gutters-159"
          disabled={false}
        >
          <div
            aria-valuenow={0}
            className="MuiCircularProgress-root-163 MuiCircularProgress-colorPrimary-166 MuiCircularProgress-static-164"
            role="progressbar"
            style={
              Object {
                "height": 40,
                "transform": "rotate(-90deg)",
                "width": 40,
              }
            }
          >
            <svg
              className="MuiCircularProgress-svg-168"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle-169 MuiCircularProgress-circleStatic-170"
                cx={44}
                cy={44}
                fill="none"
                r={19.5}
                strokeWidth={5}
                style={
                  Object {
                    "strokeDasharray": "122.522",
                    "strokeDashoffset": "122.522px",
                  }
                }
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root-172"
            style={
              Object {
                "flex": "1 1 0",
              }
            }
          >
            <span
              className="MuiTypography-root-30 MuiTypography-subheading-37 MuiListItemText-primary-175"
            >
              test2
            </span>
          </div>
        </li>
      </div>
    </div>
  </div>
</div>
`;
